File: src/main/java/kr/co/lunatalk/global/util/SecurityUtil.java
package kr.co.lunatalk.global.util;

import kr.co.lunatalk.global.exception.CustomException;
import kr.co.lunatalk.global.exception.ErrorCode;
import kr.co.lunatalk.global.security.PrincipalDetails;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

@Component
public class SecurityUtil {
	public Long getCurrentMemberId() {
		 Authentication authentication = SecurityContextHolder.getContext().getAuthentication();

		Object principal = authentication.getPrincipal();
		if (principal instanceof PrincipalDetails) {
			return Long.parseLong(((PrincipalDetails) principal).getUsername());
		}
		throw new CustomException(ErrorCode.UNAUTHORIZED);
	}

	public String getCurrentMemberRole() {
		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();

		return authentication.getAuthorities().stream()
			.map(GrantedAuthority::getAuthority)
			.findFirst()
			.orElseThrow(() -> new CustomException(ErrorCode.UNAUTHORIZED));
	}
}

File: src/main/kotlin/kr/co/lunatalk/global/util/SecurityUtil.kt
package kr.co.lunatalk.global.util

import kr.co.lunatalk.global.exception.CustomException
import kr.co.lunatalk.global.exception.ErrorCode
import kr.co.lunatalk.global.security.PrincipalDetails
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class SecurityUtil {

    fun getCurrentMemberId(): Long {
        val authentication = SecurityContextHolder.getContext().authentication

        val principal = authentication?.principal
        if (principal is PrincipalDetails) {
            return principal.username.toLong()
        }
        throw CustomException(ErrorCode.UNAUTHORIZED)
    }

    fun getCurrentMemberRole(): String {
        val authentication = SecurityContextHolder.getContext().authentication

        return authentication?.authorities
            ?.map { it.authority }
            ?.firstOrNull()
            ?: throw CustomException(ErrorCode.UNAUTHORIZED)
    }
}
